Mathlib/Topology/Separation.lean

```diff
@@ -129,6 +147,54 @@ theorem separatedNhds_iff_disjoint {s t : Set X} : SeparatedNhds s t ↔ Disjoin
 
 alias ⟨SeparatedNhds.disjoint_nhdsSet, _⟩ := separatedNhds_iff_disjoint
 
+/-- `SeparatingCover`s can be useful witnesses for `SeparatedNhds`. -/
+def SeparatingCover : Set X → Set X → Prop := fun s t ↦
+  ∃ u : ℕ → Set X, s ⊆ ⋃ n, u n ∧ ∀ n, IsOpen (u n) ∧ Disjoint (closure (u n)) t
+
+/-- Used to prove that a regular topological space with Lindelöf topology is a normal space,
+and (todo) a perfectly normal space is a completely normal space. -/
+theorem separating_covers_iff_separated_nhds {h k : Set X} :
+    SeparatingCover h k ∧ SeparatingCover k h ↔ SeparatedNhds h k := by
+  constructor
+  · rintro ⟨⟨u, u_cov, u_props⟩, ⟨v, v_cov, v_props⟩⟩
+    have open_lemma : ∀ (u₀ a : ℕ → Set X), (∀ n, IsOpen (u₀ n)) →
+      IsOpen (⋃ n, u₀ n \ closure (a n)) := fun _ _ u₀i_open ↦
+      isOpen_iUnion (fun i ↦ IsOpen.sdiff (u₀i_open i) (isClosed_closure))
+    have cover_lemma : ∀ (h₀ : Set X) (u₀ v₀ : ℕ → Set X),
+        (h₀ ⊆ ⋃ n, u₀ n) → (∀ n, Disjoint (closure (v₀ n)) h₀) →
+        (h₀ ⊆ ⋃ n, u₀ n \ closure (⋃ m ∈ {m | m ≤ n}, v₀ m)) :=
+        fun h₀ u₀ v₀ h₀_cov dis x xinh ↦ by
+      rcases h₀_cov xinh with ⟨un , ⟨n, rfl⟩ , xinun⟩
+      simp only [mem_iUnion]
+      refine ⟨n, xinun, ?_⟩
+      rw [Set.Finite.closure_biUnion (finite_le_nat n)]
+      simp_all only [disjoint_right, mem_setOf_eq, mem_iUnion, exists_false, exists_const,
+        not_false_eq_true]
+    refine ⟨
+      ⋃ n : ℕ, u n \ (closure (⋃ m ∈ {m | m ≤ n}, v m)),
+      ⋃ n : ℕ, v n \ (closure (⋃ m ∈ {m | m ≤ n}, u m)),
+      open_lemma u (fun n ↦ ⋃ m ∈ {m | m ≤ n}, v m) (fun n ↦ (u_props n).1),
+      open_lemma v (fun n ↦ ⋃ m ∈ {m | m ≤ n}, u m) (fun n ↦ (v_props n).1),
+      cover_lemma h u v u_cov (fun n ↦ (v_props n).2),
+      cover_lemma k v u v_cov (fun n ↦ (u_props n).2),
+    ?_⟩
+    rw [Set.disjoint_left]
+    rintro x ⟨un, ⟨n, rfl⟩, xinun⟩
+    simp only [mem_iUnion, mem_diff, not_exists, not_and, Decidable.not_not]
```

Comment:
Note you can be nicer to readers by saying what you will reduce to. Say the goal after this `simp only` is `foo`. Then you can replace this line with `suffices foo by simpa` (this is assuming all lemmas in your `simp` are already `simp` lemmas, otherwise you need to give them to `simpa`).

Reply:
fafa4ae

Reply:
fafa4ae

Reply:
[`fafa4ae`](https://github.com/leanprover-community/mathlib4/pull/13176/commits/fafa4ae78bddf1764eef0a4487e1b8b7a801c93a)